### ISSUE TITLE
Implement Train-time Features

### ISSUE THREAD

**Issue by andrewni2002** [andrewni420/GPU-PUSH#9]
Implement training-time features such as:
- [ ] Dropout
- [ ] Batch Normalization (?)
- [ ] Others (?)
- I think we can just have these occur after each operation in the DAG.
